### DISCUSSION TITLE
Copy button do not show "copied!" in Code Blocks

### DISCUSSION THREAD

**Discussion by evandde** [hbstack/.github#17] [Q&A] [answered]
In your website(https://hbstack.dev/), when I press copy button in a code block, I see a "Copied!" button on the center of my screen as a notification.

But in my website, it doesn't appear.

- Any other buttons are working well.
- Copying itself is working well indeed. The code is copied in my clipboard.
- The notification is not appear

For example, I made a testpage in my website (https://ffxiv.mew.kr/blog/2023/05/testpage/)

The markdown is like below

```
Summary.

<!--more-->

```html {title="public/index.html"}
<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Example HTML5 Document</title>
</head>
<body>
  <p>Test</p>
</body>
</html>
```

And my console (F12), it gives me the error below when I press copy button

```
Cannot read properties of undefined (reading 'copy_failed')
at Translator.translate (hb.c1f57f5d0cfecaa7ea304094255914e3342c65baf792251dc0da18353a312b79.js:80:39)
at hb.c1f57f5d0cfecaa7ea304094255914e3342c65baf792251dc0da18353a312b79.js:176:37
translate @ hb.c1f57f5d0cfecaa7
```

**Comment by razonyang** [comment]
Looks like a bug about missing translations, but I couldn't reproduce it on `ko` site. Could you provide a repo for reproducing it?

**Reply by evandde** [reply]
Here is my repo. (https://github.com/evandde/evanote)

As you said, the button working well in `en` language.

**Comment by evandde** [comment]
If possible, it would be great to implement the site in the following way:
For various values displayed through i18n, the default text should be in English. This means that if there is an i18n value for a specific language, it will be displayed in that language, and for missing items, the default English text will be displayed.

By doing this, even if only some i18n is available for a particular language, it would be possible to develop the site by setting that language as the default language.

Currently, it seems that when there is no i18n or only partial i18n available, it causes functional bugs. It would be beneficial to improve this situation.

**Comment by razonyang** [comment, marked as answer]
Fixed by https://github.com/hugomods/i18n-js/pull/11 and https://github.com/hugomods/code-block-panel/pull/12.

```sh
hugo mod get github.com/hugomods/code-block-panel@v0.1.2
```

You can translate the messages on https://github.com/hugomods/code-block-panel/tree/main/data/code-block-panel/i18n, will draft a new release for new translations.

**Reply by evandde** [reply]
Thx! I just made a PR.

**Reply by razonyang** [reply]
Thank you. Upgrade to `v0.2.0` should work with the new translations.
Btw, I just remember the [pagination](https://github.com/hbstack/pagination/tree/main/i18n) module that created these two days, you probably want to translate as well.
